pkg/controller/container-runtime-config/helpers.go

```diff
@@ -209,39 +218,89 @@ func updateStorageConfig(data []byte, internal *mcfgv1.ContainerRuntimeConfigura
+		return nil, fmt.Errorf("error encoding toml for CRIO log-level drop-in: %v", err)
 	}
-	if internal.LogSizeMax.Value() != 0 {
-		tomlConf.Crio.Runtime.LogSizeMax = internal.LogSizeMax.Value()
+	return newData.Bytes(), nil
+}
+
+// createCRIOdropinConfigPidsLimit decodes the data rendered from the template, merges the changes in and encodes it
+// back into a TOML format. It returns the bytes of the encoded data.This is only for when the pids-limit field is changed.
+func createCRIOdropinConfigPidsLimit(internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
+	tomlConf := new(tomlConfigCRIOPidsLimit)
+	tomlConf.Crio.Runtime.PidsLimit = internal.PidsLimit
+
+	var newData bytes.Buffer
+	encoder := toml.NewEncoder(&newData)
+	if err := encoder.Encode(*tomlConf); err != nil {
+		return nil, fmt.Errorf("error encoding toml for CRIO pids-limit drop-in: %v", err)
 	}
-	if internal.LogLevel != "" {
-		tomlConf.Crio.Runtime.LogLevel = internal.LogLevel
+	return newData.Bytes(), nil
+}
+
+// createCRIOdropinConfigLogSizeMax decodes the data rendered from the template, merges the changes in and encodes it
+// back into a TOML format. It returns the bytes of the encoded data.This is only for when the log-size-max field is changed.
+func createCRIOdropinConfigLogSizeMax(internal *mcfgv1.ContainerRuntimeConfiguration) ([]byte, error) {
+	tomlConf := new(tomlConfigCRIOLogSizeMax)
+
+	if internal.LogSizeMax.Value() == 0 {
+		return nil, fmt.Errorf("log-siz-max cannot be set to 0")
 	}
-	// For some reason, when the crio.conf file is created storage_option is not included
-	// in the file. Noticed the same thing for all fields in the struct that are of type []string
-	// and are empty. This is a dumb hack for now to ensure that cri-o doesn't blow up when
-	// overlaySize in storage.conf is set. This field being empty tells cri-o that it should take
-	// all its storage configurations from storage.conf instead of crio.conf
-	tomlConf.Crio.StorageOptions = []string{}
+	tomlConf.Crio.Runtime.LogSizeMax = internal.LogSizeMax.Value()
 
 	var newData bytes.Buffer
 	encoder := toml.NewEncoder(&newData)
 	if err := encoder.Encode(*tomlConf); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error encoding toml for CRIO log-size-max drop-in: %v", err)
 	}
-
 	return newData.Bytes(), nil
 }
 
+// We create different drop-in files for each CRI-O field that can be changed by the ctrcfg CR
+// this ensures that we don't have to rely on hard coded defaults that might cause problems
+// in future if something in cri-o or the templates used by the MCO changes
+func createCRIODropinFiles(cfg *mcfgv1.ContainerRuntimeConfig) []fileConfig {
+	var fileConfigList []fileConfig
+	ctrcfg := cfg.Spec.ContainerRuntimeConfig
+	if ctrcfg.LogLevel != "" {
+		crioLogLevelTOML, err := createCRIOdropinConfigLogLevel(cfg.Spec.ContainerRuntimeConfig)
+		if err != nil {
+			glog.V(2).Infoln(cfg, err, "error updating user changes for log-level to crio.conf.d: %v", err)
```

Comment:
Is it OK to just log? The previous implementation caused a failure.

Reply:
yes, the previous implementation was also just logging https://github.com/openshift/machine-config-operator/blob/master/pkg/controller/container-runtime-config/container_runtime_config_controller.go#L545